connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -439,8 +439,19 @@ class SparkSession private[sql] (
   }
 
   private[sql] def execute[T](plan: proto.Plan, encoder: AgnosticEncoder[T]): SparkResult[T] = {
-    val value = client.execute(plan)
-    val result = new SparkResult(value, allocator, encoder)
+    val result = if (plan.hasRoot && plan.getRoot.hasLocalRelation) {
+      // Short circuit local relation RPCs
+      val localRelation = plan.getRoot.getLocalRelation
+      new SparkResult(
+        Seq.empty[proto.ExecutePlanResponse].iterator.asJava,
```

Comment:
I am wondering if you can populate the data from the LocalRelation into an instance of `proto.ExecutePlanResponse` thus we do not need to branch the code in `SparkResult`?

Reply:
It seems a good idea.